src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs

```diff
@@ -2476,6 +2468,364 @@ public static void DCJS_RoundtrippingDateTime()
         }
     }
 
+    [Fact]
+    public static void DCJS_DateTimeFormatIsNull()
+    {
+        var settings = new DataContractJsonSerializerSettings();
+        Assert.Throws<ArgumentNullException>(() => settings.DateTimeFormat = new DateTimeFormat(null));
+        Assert.Throws<ArgumentNullException>(() => settings.DateTimeFormat = new DateTimeFormat("ddmmyyyyy", null));
+    }
+
+    [Fact]
+    public static void DCJS_NegativeDateTimeStylesTest_IncorrectFormat()
+    {
+        var dateTimeFormat = "yyyy-MM-ddTHH:mm:ss.fffK";
+        var original = new DateTime(2011, 1, 2, 3, 4, 5, 6);
+        var dcjsSettings = new DataContractJsonSerializerSettings()
+        {
+            DateTimeFormat = new DateTimeFormat(dateTimeFormat, CultureInfo.InvariantCulture)
+            {
+                DateTimeStyles = DateTimeStyles.None
+            },
+        };
+        var serializer = new DataContractJsonSerializer(typeof(DateTime), dcjsSettings);
+        var ms = new MemoryStream();
+        serializer.WriteObject(ms, original);
+        var serializedJsonValue = Encoding.UTF8.GetString(ms.ToArray());
+        serializedJsonValue = serializedJsonValue.Replace("2011", "         2011");
+        Assert.Throws<SerializationException>(() => DeserializeString<DateTime>(serializedJsonValue));
+    }
+
+    [Fact]
+    public static void DCJS_VerifyDateTimeForFormatStringDCJsonSerSettings()
+    {
+        var jsonTypes = new JsonTypes();
+        var dcjsSettings = new DataContractJsonSerializerSettings()
+        {
+            DateTimeFormat = null,
+            UseSimpleDictionaryFormat = true,
+            EmitTypeInformation = EmitTypeInformation.AsNeeded,
+            KnownTypes = new List<Type>()
+        };
+
+        var value = new DateTime(2010, 12, 1);
+        var offsetMinutes = (int)TimeZoneInfo.Local.GetUtcOffset(new DateTime()).TotalMinutes;
+        var timeZoneString = string.Format("{0:+;-}{1}", offsetMinutes, new TimeSpan(0, offsetMinutes, 0).ToString(@"hhmm"));
+
+        var baseline = $"\"\\/Date(1291161600000{timeZoneString})\\/\"";
+        if (Environment.OSVersion.Platform == PlatformID.MacOSX || Environment.OSVersion.VersionString.Contains("Windows"))
```

Comment:
Thanks for the Info. Can you please run the tests on desktop (full framework) and use the output from desktop as baseline? And then skip the test on the platforms where the test fails.